x-pack/plugins/apm/public/hooks/useEnvironments.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { useFetcher } from './useFetcher';
+import {
+  ENVIRONMENT_NOT_DEFINED,
+  ENVIRONMENT_ALL
+} from '../../common/environment_filter_values';
+import { useUrlParams } from './useUrlParams';
+
+export const ALL_OPTION = {
+  value: ENVIRONMENT_ALL,
+  text: i18n.translate('xpack.apm.environment.allLabel', {
+    defaultMessage: 'All'
+  })
+};
+
+function getEnvironmentOptions(environments: string[]) {
+  const environmentOptions = environments
+    .filter(env => env !== ENVIRONMENT_NOT_DEFINED)
+    .map(environment => ({
+      value: environment,
+      text: environment
+    }));
+
+  return [ALL_OPTION, ...environmentOptions];
+}
+
+export function useEnvironments() {
+  const { urlParams } = useUrlParams();
+  const { serviceName, start, end } = urlParams;
+  const { data: environments = [], status = 'loading' } = useFetcher(
+    callApmApi => {
```

Comment:
we can just use callApmApi directly here no?

Reply:
yes!!!

Reply:
Yes, we could, but I'm refactored [EnvironmentFilter](https://github.com/cauemarcondes/kibana/blob/environment-alert/x-pack/plugins/apm/public/components/shared/EnvironmentFilter/index.tsx#L73) component to use this hook too, and it needs the status which is not available when calling `callApmApi` directly.

Reply:
What I meant was that you don't have to use the function argument from `useFetcher`, now that `callApmApi` is a global again. We can probably remove it entirely (from the type and from the runtime code) in `useFetcher`, there might be other instances where we are still doing this.

Reply:
ok I got your point

Reply:
I removed it from `useEnvironment`, and I'm going to create a new tech debit issue to remove it from `useFetcher`.

Reply:
https://github.com/elastic/kibana/issues/66352